modules/azure/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/azure/AzureBlobStorage.scala

```diff
@@ -76,7 +76,7 @@ class AzureBlobStorage[F[_]: Async] private (store: AzureStore[F], path: AzureBl
       .map(authority => Url(path.scheme, authority, Path(path.extractRelative(input))))
 
   private def createBlobObject(url: Url[AzureBlob]) = {
-    val key = path.fullPath.withKey(url.path.relative.show)
+    val key = path.root.append(path.containerName).withKey(url.path.relative.show)
```

Comment:
I've fixed a few bugs in here AzureBlobStorage. It might be good to have some unit tests for this class. I was planning to take a stab for it today but I couldn't have a chance. @pondzix could you give it a go if you have availability next week ? No worries if you can't, I can add some tests when I get back following week.